src/app/components/repo-list-component/repo-list.component.ts

```diff
@@ -1,14 +1,15 @@
+import { DataHelpers } from "src/app/common/data.helpers";
 import { DatasetInfo } from "./../../interface/navigation.interface";
 import { NavigationService } from "./../../services/navigation.service";
-import { Component, EventEmitter, Input, Output } from "@angular/core";
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from "@angular/core";
 import { ModalService } from "../modal/modal.service";
-import { DataHelpersService } from "src/app/services/datahelpers.service";
 import { Dataset } from "src/app/api/kamu.graphql.interface";
 
 @Component({
     selector: "app-repo-list",
     templateUrl: "./repo-list.component.html",
     styleUrls: ["./repo-list.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
@zaychenko-sergei should we require all components to use `OnPush` change detection?

From my shallow understanding the `Default` strategy seems like an absolute performance killer and perhaps even a potential source of infinite feedback update loops. So enforcing `OnPush` might be a good idea?

Can we:
- configure app to use it globally?
- enforce it via base class?

Reply:
This makes me want to go back to Svelte :)

Reply:
@sergiimk We have that item in the FE tasks list, but this needs a careful evaluation. `OnPush` is only a valid option, when the state of the component fully determined by the values of `@Input`s. Since we have subscription-driven components, this needs a careful review and potentially manual tweaks to trigger detection of changes via [`ChangeDetectorRef`.](https://angular.io/api/core/ChangeDetectorRef). There is also a popular pattern called [`Smart and Dump Components`](https://medium.com/@mrahmedkhan019/smart-dumb-components-in-angular-e1d6dbd6edff)